sklearn/ensemble/gradient_boosting.py

```diff
@@ -1073,7 +1114,31 @@ def _init_decision_function(self, X):
         if X.shape[1] != self.n_features:
             raise ValueError("X.shape[1] should be {0:d}, not {1:d}.".format(
                 self.n_features, X.shape[1]))
-        score = self.init_.predict(X).astype(np.float64)
+            # init predictions
+
+        if hasattr(self.init_, 'predict_proba'):
+            eps = np.finfo(X.dtype).eps
+            score = self.init_.predict_proba(X) + eps
+            if self.n_classes == 2:
+                score = np.log(score[:,1] / score[:,0])
```

Comment:
I'm using the log odds here, like the LogOdds estimator class returns in its `predict` function. Is this incorrect?